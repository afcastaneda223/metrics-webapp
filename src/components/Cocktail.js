import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getID } from '../redux/cocktail/cocktail';

function Cocktail() {
  const dispatch = useDispatch();
  const myDetailID = useSelector((state) => state.details.detailID);
  const fetchApi = `https://www.thecocktaildb.com/api/json/v1/1/lookup.php?i=${myDetailID}`;

  const myDetailsID = useSelector((state) => state.cocktail);

  useEffect(() => {
    const apiID = async () => {
      const fetchid = await fetch(fetchApi);
      const detailId = await fetchid.json();
      return dispatch(getID(detailId));
    };
    apiID();
  }, []);

  return (
    <div>
      { myDetailsID.map((details) => (
        <div className="card bg2 border-0" key={details.strDrink}>
          <img src={details.image} className="p-5 bg1 border-0" alt="..." />
          <div className="card-body bg2">
            <h5 className="card-title fw-bold">{details.name}</h5>
            <p>{details.instructions}</p>
            {details.ingredients.map((ingredient) => (
              <div className="list-group" key={details.id}>
                { ingredient != null ? <li className="list-group-item list-group-item-action rounded-0">{ingredient}</li> : null }
              </div>
            ))}
          </div>
        </div>
      ))}
    </div>
  );
}

export default Cocktail;
