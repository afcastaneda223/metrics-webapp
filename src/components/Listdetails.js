import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getDetail } from '../redux/details/details';

function Listdetails() {
  const dispatch = useDispatch();
  const myCocktailIngredient = useSelector((state) => state.cocktails.ingredient);
  const fetchApi = `https://www.thecocktaildb.com/api/json/v1/1/filter.php?i=${myCocktailIngredient}`;

  const myDetailsArray = useSelector((state) => state.details);

  useEffect(() => {
    const apiDetails = async () => {
      const fetchDetail = await fetch(fetchApi);
      const details = await fetchDetail.json();
      return dispatch(getDetail(details));
    };
    apiDetails();
  }, []);

  return (
    <div className="m-0 p-0">
      {myDetailsArray.map((detail) => (
        <div className="card main-c2" key={detail.id}>
          <div className="row">
            <div className="col align-self-center">
              <div className="card-body">
                <p className="card-title fw-bold">{detail.name}</p>
              </div>
            </div>
            <div className="col d-flex justify-content-end">
              <img src={detail.image} className="detailsimg" alt="..." />
            </div>
          </div>
        </div>
      ))}
    </div>
  );
}

export default Listdetails;
