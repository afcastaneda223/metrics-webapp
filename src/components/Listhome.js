import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getIngredients, pickIngredient } from '../redux/home/home';

function Listhome() {
  const dispatch = useDispatch();
  const fetchApi = 'https://www.thecocktaildb.com/api/json/v1/1/list.php?i=list';

  const myCocktailArray = useSelector((state) => state.home.array);

  useEffect(() => {
    const apiCocktail = async () => {
      const fetchCocktail = await fetch(fetchApi);
      const cocktails = await fetchCocktail.json();
      return dispatch(getIngredients(cocktails));
    };
    if (myCocktailArray.length === 0) {
      apiCocktail();
    }
  }, []);

  const selectCocktail = (e) => {
    const x = e.target.id;
    dispatch(pickIngredient(x));
  };

  return (
    <Link to="/details" className="m-0 p-0 white container" onClick={selectCocktail}>
      <div>
        <div className="row row-cols-2 border-0 m-0 p-0 ">
          { myCocktailArray.map((cocktail) => (
            <div className="col m-0 p-0" key={cocktail.Ingredient}>
              <div className="card border-0 main-c2 rounded-0">
                <img src={cocktail.url} className="card-img opacity-75 rounded-0" alt="..." />
                <div className="card-img-overlay h-100 d-flex flex-column justify-content-end" id={cocktail.Ingredient}>
                  <h5 className="card-title text-center fw-bolder" id={cocktail.Ingredient}>{cocktail.Ingredient}</h5>
                </div>
              </div>
            </div>
          ))}
        </div>
      </div>
    </Link>
  );
}

export default Listhome;
