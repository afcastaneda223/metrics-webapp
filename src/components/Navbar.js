import React from 'react';
import { NavLink } from 'react-router-dom';

function Navbar() {
  return (
    <nav className="navbar navbar-expand">
      <div className="container-fluid main-navbar">
        <NavLink to="/">
          <i className="fas fa-chevron-left white" />
        </NavLink>
        <ul className="navbar-nav text-end">
          <li className="m-2 white" key="home">
            <i className="fas fa-microphone" />
          </li>
          <li className="m-2 white" key="details">
            <i className="fas fa-cog" />
          </li>
        </ul>
      </div>
    </nav>
  );
}

export default Navbar;
