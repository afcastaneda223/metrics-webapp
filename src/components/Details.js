import React from 'react';
import { useSelector } from 'react-redux';

import Listdetails from './Listdetails';

function Details() {
  const myCocktailArray = useSelector((state) => state.cocktails.ingredient);

  return (
    <>
      <div>
        <div className="card border-0">
          <div className="row g-0 main-c1">
            <div className="col">
              <img src={`https://www.thecocktaildb.com/images/ingredients/${myCocktailArray}.png`} className="img-fluid rounded-0 opacity-75" alt="..." />
            </div>
            <div className="col align-self-center">
              <div className="card-body">
                <h5 className="card-title fw-bolder text-uppercase">{myCocktailArray}</h5>
              </div>
            </div>
          </div>
        </div>
      </div>
      <h6 className="text-uppercase">
        COCKTAILS WITH
        {' '}
        {myCocktailArray}
      </h6>
      <Listdetails />
    </>
  );
}

export default Details;
