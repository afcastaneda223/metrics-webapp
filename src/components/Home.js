import React from 'react';
import { useSelector } from 'react-redux';
import Listhome from './Listhome';
import image1 from './assets/cocktails.png';

function Home() {
  const myCocktailArray = useSelector((state) => state.home.array);
  return (
    <>
      <div className="card border-0">
        <div className="row g-0 main-c1">
          <div className="col">
            <img src={image1} className="img-fluid rounded-0" alt="..." />
          </div>
          <div className="col align-self-center">
            <div className="card-body">
              <h5 className="card-title fw-bolder">COCKTAILS</h5>
              <p className="card-text fw-light">
                { myCocktailArray.length }
                {' '}
                ingredients total
              </p>
            </div>
          </div>
        </div>
      </div>
      <div>
        <h6>COCKTAILS BY INGREDIENT</h6>
        <Listhome />
      </div>
    </>
  );
}

export default Home;
