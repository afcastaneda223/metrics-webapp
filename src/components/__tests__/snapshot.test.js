import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import {
  BrowserRouter,
  BrowserRouter as Router,
} from 'react-router-dom';
import Details from '../Details';
import Home from '../Home';
import Cocktail from '../Cocktail';
import Listhome from '../Listhome';
import Listdetails from '../Listdetails';
import Navbar from '../Navbar';
import store from '../../redux/configureStore';

describe('Render Navbar', () => {
  it('Renders the navbar', () => {
    const header = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    ).toJSON();
    expect(header).toMatchSnapshot();
  });
});

describe('Render Home page', () => {
  it('Renders the Home page', () => {
    const home = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(home).toMatchSnapshot();
  });
});

describe('Render Listhome page', () => {
  it('Renders the Listhome page', () => {
    const listhome = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Listhome />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(listhome).toMatchSnapshot();
  });
});

describe('Render Details page', () => {
  it('Renders the Details page', () => {
    const details = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(details).toMatchSnapshot();
  });
});

describe('Render Listdetails page', () => {
  it('Renders the Listdetails page', () => {
    const listdetails = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Listdetails />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(listdetails).toMatchSnapshot();
  });
});

describe('Render Cocktail page', () => {
  it('Renders the Cocktail page', () => {
    const cocktail = TestRenderer.create(
      <Provider store={store}>
        <Cocktail />
      </Provider>,
    ).toJSON();
    expect(cocktail).toMatchSnapshot();
  });
});
