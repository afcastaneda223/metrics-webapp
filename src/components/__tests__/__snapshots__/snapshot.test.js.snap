// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Cocktail page Renders the Cocktail page 1`] = `<div />`;

exports[`Render Details page Renders the Details page 1`] = `
Array [
  <div>
    <div
      className="card border-0"
    >
      <div
        className="row g-0 main-c1"
      >
        <div
          className="col"
        >
          <img
            alt="..."
            className="img-fluid rounded-0 opacity-75 detailsimg"
            src="https://www.thecocktaildb.com/images/ingredients/null.png"
          />
        </div>
        <div
          className="col align-self-center"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title fw-bolder text-uppercase"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <h6
    className="text-uppercase m-0 p-0"
  >
    COCKTAILS WITH
     
  </h6>,
  <a
    className="m-0 p-0 white container"
    href="/cocktail"
    onClick={[Function]}
  >
    <div
      className="m-0 p-0"
    />
  </a>,
]
`;

exports[`Render Home page Renders the Home page 1`] = `
Array [
  <div
    className="card border-0"
  >
    <div
      className="row g-0 main-c1"
    >
      <div
        className="col"
      >
        <img
          alt="..."
          className="img-fluid rounded-0 detailsimg"
          src="cocktails.png"
        />
      </div>
      <div
        className="col align-self-center"
      >
        <div
          className="card-body"
        >
          <h5
            className="card-title fw-bolder"
          >
            COCKTAILS
          </h5>
          <p
            className="card-text fw-light"
          >
            0
             
            ingredients total
          </p>
        </div>
      </div>
    </div>
  </div>,
  <div>
    <h6
      className="m-0 p-0"
    >
      COCKTAILS BY INGREDIENT
    </h6>
    <a
      className="m-0 p-0 white container"
      href="/details"
      onClick={[Function]}
    >
      <div>
        <div
          className="row row-cols-2 border-0 m-0 p-0 "
        />
      </div>
    </a>
  </div>,
]
`;

exports[`Render Listdetails page Renders the Listdetails page 1`] = `
<a
  className="m-0 p-0 white container"
  href="/cocktail"
  onClick={[Function]}
>
  <div
    className="m-0 p-0"
  />
</a>
`;

exports[`Render Listhome page Renders the Listhome page 1`] = `
<a
  className="m-0 p-0 white container"
  href="/details"
  onClick={[Function]}
>
  <div>
    <div
      className="row row-cols-2 border-0 m-0 p-0 "
    />
  </div>
</a>
`;

exports[`Render Navbar Renders the navbar 1`] = `
<nav
  className="navbar navbar-expand"
>
  <div
    className="container-fluid main-navbar"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <i
        className="fas fa-chevron-left white"
      />
    </a>
    <ul
      className="navbar-nav text-end"
    >
      <li
        className="m-2 white"
      >
        <i
          className="fas fa-microphone"
        />
      </li>
      <li
        className="m-2 white"
      >
        <i
          className="fas fa-cog"
        />
      </li>
    </ul>
  </div>
</nav>
`;
